### PR TITLE
[jit] Remove NoneGenerator

### PR DESCRIPTION
**Pull request by driazati**

### CODE REVIEW

**Review by zdevito — APPROVED** [review]
LGTM. I think the implicit conversion from Generator -> None can be removed as well now.

File: torch/csrc/jit/register_prim_ops.cpp [hunk 1]
```diff
@@ -259,14 +259,6 @@ RegisterOperators reg({
           return 0;
         };
       }),
-    Operator(
```
Comment (apaszke) [inline torch/csrc/jit/register_prim_ops.cpp]: Can we really keep deleting operators like that? Won't this break existing checkpoints??

Comment (driazati) [reply torch/csrc/jit/register_prim_ops.cpp]: Probably, should we instead add this back into a separate list of legacy/deprecated ops?

Comment (zdevito) [reply torch/csrc/jit/register_prim_ops.cpp]: I think it should work fine if we translate calls to prim::NoneGenerator into prim::None when we load the model in import_methods. But it should be tested to see that an old network works.

**Pull request closed without merging**